RallyHereIntegration/Source/RallyHereIntegration/Private/RH_SessionHelpers.h

```diff
@@ -852,3 +869,29 @@ class FRH_SessionJoinByPlatformIdHelper : public FRH_SessionLookupHelper
 
 	FRH_OnSessionUpdatedDelegateBlock Delegate;
 };
+
+
+// Wrapper around a delete that will remove the session locally once the delete is completed, without requiring a poll on the session
+class FRH_SessionLeaveHelper : public FRH_SessionRequestAndModifyHelper<RallyHereAPI::Traits_LeaveSessionByIdSelf>
+{
+public:
+	typedef RallyHereAPI::Traits_LeaveSessionByIdSelf BaseType;
+
+	FRH_SessionLeaveHelper(FRH_SessionOwnerPtr InSessionOwner, const FString& InSessionId, FRH_OnSessionUpdatedDelegateBlock InDelegate = FRH_OnSessionUpdatedDelegateBlock(), int32 InPriority = DefaultRallyHereAPIPriority)
+		: FRH_SessionRequestAndModifyHelper(InSessionOwner, InSessionId, InDelegate, InPriority)
+	{
+	}
+
+	virtual void OnSessionPollComplete(bool bSuccess, bool bResetTimer)
+	{
+		// ignore success flag on the poll (it can succeed or fail depending on the session state, we just want to verify that it removed the session from the owner)
```

Comment:
Are you sure it is okay not to call the base function anywhere?  I suspect you are correct, but want to point it out

Reply:
I was considering it, but the base function internally completes the helper with the status being whether or not the session exists.  If it exists, it could potentially give a false positive (since it would indicate that the delete succeeded).